crates/rome_js_analyze/src/semantic_analyzers/nursery/no_constant_condition.rs

```diff
@@ -106,6 +112,16 @@ impl Rule for NoConstantCondition {
         }
 
         let test = conditional_stmt.test()?;
+        // Ignore `while (true) { ... }`
+        if let ConditionalStatement::JsWhileStatement(_) = conditional_stmt {
+            if let Some(test) = test.as_any_js_literal_expression() {
+                if let Some(test) = test.as_js_boolean_literal_expression() {
+                    if test.value_token().ok()?.kind() == JsSyntaxKind::TRUE_KW {
+                        return None;
+                    }
+                }
+            }
```

Comment:
You can chain all these and avoid at least two `if let Some` by using `.and_then`